torch/distributed/_tensor/placement_types.py

```diff
@@ -37,7 +37,17 @@ def is_partial(self) -> bool:
 
 @dataclass(frozen=True)
 class Shard(Placement):
-    # shard placement, shard on a dim
+    """
+    The ``Shard(dim)`` placement describes the DTensor sharding on a specified
```

Comment:
I think somehow this description should clarify/mention that the `dim` in `Shard(dim)` is the _tensor_ dim across which sharding happens, not the `DeviceMesh` dimension. (I think this might be a common source of confusion.)

Reply:
yeah good point! updating